packages/synapse-react-client/src/components/widgets/query-filter/EnumFacetFilter/EnumFacetFilterUI.tsx

```diff
@@ -0,0 +1,277 @@
+  /* Whether the 'All' checkbox should be selected. */
+  allIsSelected: boolean // allIsSelected cannot be derived from facetValues[*].isSelected. See PORTALS-2680
+  /* The EnumFacetFilter can be rendered as a collapsible panel or a dropdown. Default is `'Collapsible'` */
+  containerAs?: 'Collapsible' | 'Dropdown'
+  /* If `containerAs` is a dropdown, the dropdown can be either an IconButton or a SelectBox. Default is `'Icon'` */
+  dropdownType?: 'Icon' | 'SelectBox'
+  /* If true, hides the button to collapse the panel (only shown if `containerAs` is 'Collapsible'). Default is false.*/
+  hideCollapsible?: boolean
+  /* Called when an unselected value is selected */
+  onAddValueToSelection: (value: string) => void
+  /* Called when a selected value is unselected */
+  onRemoveValueFromSelection: (value: string) => void
+  /* Called when all values should be unselected (e.g. 'All' is clicked) */
+  onRemoveAllFacetSelections: () => void
+}
+
+function removeWhitespace(value: string): string {
+  return value.replace(/\s/g, '').toLowerCase()
+}
+
+export default function EnumFacetFilterUI(props: EnumFacetFilterUIProps) {
+  const {
+    allIsSelected,
+    containerAs = 'Collapsible',
+    dropdownType = 'Icon',
+    hideCollapsible = false,
+    facetValues,
+    onAddValueToSelection,
+    onRemoveValueFromSelection,
+    facetTitle,
+    onRemoveAllFacetSelections,
+  } = props
+  const [toggleShowAll, setToggleShowAll] = useState<boolean>(false)
+  const [isCollapsed, setIsCollapsed] = useState<boolean>(false)
+  const [showSearch, setShowSearch] = useState<boolean>(false)
+  const [searchTerm, setSearchText] = useState<string>('')
+
+  // e.g. we show a subset of facet values before hiding the rest behind "Show More"
+  // If the user has selected a value that would normally be hidden, always show all of the values
+  const valueIndexedGreaterThanMaxToShowIsSelected =
+    facetValues.length > MAX_ENUMERATION_VALUES_TO_SHOW &&
+    facetValues
+      .slice(MAX_ENUMERATION_VALUES_TO_SHOW)
+      .some(({ isSelected }) => isSelected)
+  const showAll =
+    toggleShowAll || showSearch || valueIndexedGreaterThanMaxToShowIsSelected
+
+  const facetValuesToShow = useMemo(() => {
+    if (showSearch && searchTerm.length > 0) {
+      return facetValues.filter(({ displayText }) => {
+        return (
+          displayText.toLowerCase().indexOf(searchTerm.trim().toLowerCase()) >
+          -1
+        )
+      })
+    }
+    if (showAll) {
+      return facetValues
+    }
+    return facetValues.slice(0, MAX_ENUMERATION_VALUES_TO_SHOW)
+  }, [facetValues, searchTerm, showSearch, showAll])
+
+  const textInput: React.RefObject<HTMLInputElement> = React.createRef()
+
+  const isDropdown = containerAs === 'Dropdown'
+  const content = (
+    <div className={isDropdown ? 'EnumFacetFilter__dropdown_menu' : ''}>
+      <div className="EnumFacetFilter__checkboxContainer--forAll">
+        <TextField // Search Filter Text
```

Comment:
Changed to use MUI

![image](https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/17580037/964d1bcf-7110-4f5d-9276-07bff1c8f984)